kythe/go/platform/analysis/local/local.go

```diff
@@ -140,3 +144,98 @@ type kzipFetcher struct{ r *kzip.Reader }
 
 // Fetch implements the required method of analysis.Fetcher.
 func (k kzipFetcher) Fetch(_, digest string) ([]byte, error) { return k.r.ReadAll(digest) }
+
+// Analyzer provides an implementation of the analysis.CompilationAnalyzer
+// interface that works by saving the compilation unit to a temporary kzip file
+// and invoking an indexer binary on it.
+type Analyzer struct {
+	cmd     []string
+	fetcher analysis.Fetcher
+}
+
+// NewAnalyzer creates an analyzer that will execute the given command when
+// Analyze() is called. The filepath to a kzip is appended to the command, so if
+// given ["my_indexer", "--index"], the final executed command will be
+// ["my_indexer", "--index", "some/file.kzip"]. The indexer command should emit
+// length-delimited Entry protos to stdout as described in
+// //kythe/go/platform/delimited.
+func NewAnalyzer(cmd []string, fetcher analysis.Fetcher) *Analyzer {
+	if len(cmd) == 0 {
+		panic("parameter 'cmd' must not be empty")
+	}
+	return &Analyzer{cmd: cmd, fetcher: fetcher}
+}
+
+// Analyze implements the analysis.CompilationAnalyzer interface.
+func (a *Analyzer) Analyze(ctx context.Context, req *apb.AnalysisRequest, f analysis.OutputFunc) error {
+	tmpkzip, err := saveSingleUnitKzip(ctx, req.Compilation, a.fetcher)
+	if err != nil {
+		return fmt.Errorf("creating tmp kzip: %v", err)
+	}
+	defer os.Remove(tmpkzip)
+
+	// Run indexer binary
+	out, err := exec.Command(a.cmd[0], append(a.cmd[1:], tmpkzip)...).Output()
```

Comment:
Consider using `exec.CommandContext` here, and plumbing in `ctx`. That way if the caller cancels the context, the subcommand will get terminated and this will return without any extra effort on our part. 👍

Reply:
Thanks, I wasn't aware of `CommandContext`, but that works great here.